packages/open-truss/src/configuration/engine-v1.tsx

```diff
@@ -1,24 +1,33 @@
 import React from 'react'
 import { z } from 'zod'
 import { YamlObjectShape, YamlShape } from '../utils/yaml'
-import { type COMPONENTS, type ReactTree, type RenderingEngine } from './apply'
+import {
+  type OpenTrussComponentExports,
+  type RenderingEngine,
+  type ReactTree,
+  type COMPONENTS,
+} from './apply'
 
-type Components = React.JSX.Element
+const DataShape = YamlShape.optional()
 
 const FrameBase = z.object({
+  frame: z.null(), // used only to make configs more readable
   view: z.object({
     component: z.string(),
-    props: YamlObjectShape,
+    props: YamlObjectShape.optional(),
   }),
-  data: YamlShape,
+  data: DataShape,
 })
 
-type FrameType = z.infer<typeof FrameBase> & {
-  frames: FrameType[]
+export type FrameType = z.infer<typeof FrameBase> & {
+  frames?: FrameType[]
 }
 
 const FrameV1Shape: z.ZodType<FrameType> = FrameBase.extend({
-  frames: z.lazy(() => FrameV1Shape).array(),
+  frames: z
+    .lazy(() => FrameV1Shape)
```

Comment:
I'm glad you figured out how to do recursive types, since we love those so much!